orchestration/task_dispatcher.py

```
@@ -1085,6 +1299,42 @@ def create_dynamic_agent(self, agent_spec: dict) -> bool:
+        try:
+            mock_dir = Path(tempfile.gettempdir()) / "worldarchitect_ai"
+            mock_dir.mkdir(parents=True, exist_ok=True)
+            mock_path = mock_dir / "mock_claude.sh"
+
+            # Simple shim that echoes the call for logging and exits successfully
+            script_contents = """#!/usr/bin/env bash
+echo "[mock claude] $@"
+exit 0
+"""
+            mock_path.write_text(script_contents, encoding="utf-8")
+            os.chmod(mock_path, 0o755)
```

Ruff (0.13.3)

`os.chmod` setting a permissive mask `0o755` on file or directory

(S103)